src/pages/bucketlist_items.jsx

```diff
@@ -0,0 +1,292 @@
+                records_length: ItemStore.records_length                
+            });
+
+            // Set information for the pagination component (number of pages)
+            if(this.state.bucketlist_items.length === 0){
+                this.setState({
+                    status: true
+                })
+            } else {
+                this.setState({
+                    status: false
+                })
+            }
+            if(this.state.records_length !== 0){
+                let items = Math.floor(this.state.records_length / this.state.limit);
+                if(this.state.records_length > 10){
+                    items = items + 1;
+                } else {
+                    items = 0;
+                }
+                this.setState({
+                    items: items
+                })
+            }
+        });
+    }
+
+    // store values in the state when they are put in the form
+    handleChange(event){
+        this.setState({
+            [event.target.id]: event.target.value
+        });
+    }
+
+    // submit the values in the form (which have been saved in the state)
+    handleSubmit(e){
+        e.preventDefault();
+        const payload = {
+            "name" : this.state.name,
+            "description": this.state.description
+        }
+        if(this.state.editid){
+            itemactions.editItem(this.state.editid, payload);
+            this.close();
+        }else{
+            //throw in the token and call for a create-item action
+            itemactions.createItem(payload);
+            this.setState({
+                active_page: 1
+            })
+        }
+    }
+
+    handleDelete(id){
+        itemactions.deleteItem(String(id), {page: this.state.active_page-1});
+    }
+
+    handleSearch(event){
+        itemactions.searchItem({'q': event.target.value});
+        this.setState({
+            active_page: 1
+        })
+    }
+
+    selectPage(pageNumber) {
+        this.setState({
+            active_page: pageNumber
+        });
+        ItemStore.retrieveItems(this.state.bucketlist_id, this.state.bucketlist_name, {'page':pageNumber-1});
+    }
+
+    handleLogout(){
+        authactions.logout();
+        this.setState({
+            redirect: "logout",
+            url: "/"
+        })
+    }
+
+    handleEdit(id, name, description, cond){
+        //open modal with the edit-bucketlist form
+        this.setState({
+            show_modal: cond,
+            editid: id,
+            name: name,
+            description: description
+        });
+    }
+
+    // close the modal when done
+    close(){
+        this.setState({
+            show_modal: false,
+            editid: null,
+            name: "",
+            description: ""
+        });
+        document.getElementById("create-item-form").reset();        
+    }
+
+    render(){
+        let bucketlistItems = this.state.bucketlist_items;
+        console.log("captured items: ", bucketlistItems);
```

Comment:
why this `console.log`?

Reply:
It was to track the items that were saved in the store and passed to the component. Mostly for development purposes, it's not needed in the production build